enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,1029 @@
+should apply to all subnets specified in `platform.aws.vpc.subnets`.
+
+###### Multiple Load Balancer Subnets in the Same AZ Validation
+
+The installer must reject multiple `IngressControllerLB` role subnets in the same AZ as this will be
+rejected by the AWS CCM.
+
+###### The Old and New Subnets Fields Cannot be Specified Together Validation
+
+Since `platform.aws.subnets` will be deprecated, the installer must not allow `platform.aws.vpc.subnets` and
+`platform.aws.subnets` to both be specified.
+
+###### Maximum of 10 IngressController Subnets Validation
+
+Since the IngressController's API [only allows 10 subnets](https://github.com/openshift/api/blob/ee5bb7eaf6b6638d4e3b33ba4ff0834212cdb75d/operator/v1/types_ingress.go#L564),
+the installer must not allow more than 10 IngressController subnets as well.
+
+###### Consistent Cluster Scope with IngressControllerLB Subnets Validation
+
+The installer must not allow **any** public IngressControllerLB subnets for private clusters (`installconfig.publish: Internal`)
+as this is a security risk as discussed in the [Do Not Use Untagged Subnets User Stories](#do-not-use-untagged-subnets-user-stories) section.
+
+Conversely, the installer must now allow **any** private IngressControllerLB subnets for public clusters
+(`installconfig.publish: External`) as the public IngressController will not function with private subnets.
+
+###### Consistent Cluster Scope with ControlPlaneLB Subnets Validation
+
+The installer must not allow **any** public subnets to be specified with the ControlPlaneInternalLB role and must
+not allow **any** private subnets to be specified with the ControlPlaneExternalLB role.
+
+Private clusters (`installconfig.publish: Internal`) should reject an install-config containing
+any ControlPlaneExternalLB roles, as only an internal control plane load balancer will be created.
+
+###### Reject BYO VPC Installations that Contain Untagged Subnets
+
+As described in [Rejecting Untagged Subnets Proposal](#rejecting-untagged-subnets-proposal), the installer
+must not allow installation into an existing VPC if it contains subnets lacking the `kubernetes.io/cluster/<cluster-id>`
+tag when subnets roles are not specified. The error message should instruct the user to add a
+`kubernetes.io/cluster/unmanaged` tag to any subnets they wish to exclude from this cluster installation or
+configure the subnets with the Cluster or IngressController role in `platform.aws.vpc.subnets`.
+
+###### Reject IngressControllers or ControlPlaneLB AZs that do not match ClusterNode AZs
+
+When manual role selection is specified, the installer must ensure that the AZs for the IngressController
+(defined by the IngressControllerLB subnets) and control plane (defined by the ControlPlaneExternalLB and
+ControlPlaneInternal subnets) match the cluster AZs (defined by the ClusterNode subnets). The installer must
+reject installations where the AZs are not equal.
```

Comment:
For completeness, we will also need to ensure all azs specified in the machine pools `installConfig.compute` and `installConfig.controlPlane` have specified subnets.